### ISSUE TITLE
Integer values of zero are ignored on Export

### ISSUE THREAD

**Issue by mjsqu** [dfinke/ImportExcel#445]
The following code produces a worksheet with the values 1,2. It completely ignores the zeroes:

`@(0,0,0,0,1,0,0,2,0)   | Export-Excel -Path "C:\temp\one_to_ten3.xlsx" -WorksheetName "New Data" -Show -ClearSheet -Verbose `

I found the issue by running this:

`-1..10 | Export-Excel -Path "C:\temp\one_to_ten3.xlsx" -WorksheetName "New Data" -Show -ClearSheet -Verbose`

Which properly iterated through the range, but didn't export the zero value

**Comment by dfinke** [comment]
Thanks for logging the issue.

# Also doesn't not work.

```powershell
$xlfile = "$env:TEMP\test.xlsx"
rm $xlfile -ErrorAction SilentlyContinue

[pscustomobject]@{id=-1..10}|Export-Excel $xlfile -Show
```

# Does work

```powershell
ConvertFrom-Csv @"
ID
-1
0
1
2
3
4
5
6
7
8
9
10
"@ | Export-Excel
```

**Comment by mjsqu** [comment]
I'm not sure that's the same issue, this:

`[pscustomobject]@{id=-1..10} | %{ Write-Host $_ }`

Only returns one object, which only contains the value '-1'. Technically you've asked for a PSCustomObject containing a dictionary with an array as its value.

### These two approaches work OK:

```
-1..10 | % -Begin {$arraylist = New-Object System.Collections.ArrayList} -Process {
    $arraylist += [pscustomobject]@{id=$_}
}


$xlfile = "$env:TEMP\test.xlsx"
rm $xlfile -ErrorAction SilentlyContinue

$arraylist |Export-Excel $xlfile -Show 
```

### Checking if a standard array works - YES

```
-1..10 | % -Begin {$array = @()} -Process {
    $array += [pscustomobject]@{id=$_}
}


$xlfile = "$env:TEMP\test.xlsx"
rm $xlfile -ErrorAction SilentlyContinue

$array |Export-Excel $xlfile -Show

```
### This is the code that causes a problem (having the array elements as integers)

```
-1..10 | % -Begin {$array = @()} -Process {
    $array += $_
}


$xlfile = "$env:TEMP\test.xlsx"
rm $xlfile -ErrorAction SilentlyContinue

$array |Export-Excel $xlfile -Show 
```

### Trying Strings - This works:

```
-1..10 | % -Begin {$array = @()} -Process {
    $array += "$_"
}


$xlfile = "$env:TEMP\test.xlsx"
rm $xlfile -ErrorAction SilentlyContinue

$array |Export-Excel $xlfile -Show 
```

So in summary, it's something to do with the code handling an integer value of zero as a completely absent value.

**Comment by dfinke** [comment]
Just leaving notes for things to look into. Thanks.

**Comment by mjsqu** [comment]
No worries, overall it's a wonderful tool. Just trying to narrow down the issue and produce a more isolated problem to solve. Thank you.

**Comment by dfinke** [comment]
Yeah, this is great, the more examples the better. Be great to wire these up as unit tests.

**Comment by jhoneill** [comment]
Yeah … remember this  https://jamesone111.wordpress.com/2018/07/30/on-powershell-parameters/ 
We have  `if ($TargetData) { `  (line 684)
So if  $TargetData evaluates to False (so false, 0, empty string etc) it won't get processed. 

 changing that line to `if ($PSBoundParameters.ContainsKey(TargetData)) {`
should fix it.
